static/src/javascripts/projects/commercial/modules/prebid/appnexus.spec.js

```diff
@@ -0,0 +1,276 @@
+// @flow
+import config from 'lib/config';
+
+import {
+    _,
+    getAppNexusDirectBidParams,
+    getAppNexusServerSideBidParams,
+} from './appnexus';
+
+import type { PrebidSize } from './types';
+
+import { getBreakpointKey as getBreakpointKey_ } from './utils';
+
+jest.mock('common/modules/commercial/build-page-targeting', () => ({
+    buildAppNexusTargeting: () => 'someTestAppNexusTargeting',
+    buildAppNexusTargetingObject: () => 'someAppNexusTargetingObject',
+    buildPageTargeting: () => 'pageTargeting',
+}));
+
+jest.mock('./utils', () => {
+    // $FlowFixMe property requireActual is actually not missing Flow.
+    const original = jest.requireActual('./utils');
+    return {
+        ...original,
+        getBreakpointKey: jest.fn(),
+        isInAuRegion: jest.fn(),
+    };
+});
+
+const {
+    getAppNexusPlacementId,
+    getAppNexusInvCode,
+    getAppNexusDirectPlacementId,
+} = _;
+
+const getBreakpointKey: any = getBreakpointKey_;
+
+/* eslint-disable guardian-frontend/no-direct-access-config */
+const resetConfig = () => {
+    config.set('switches.prebidAppnexus', true);
+    config.set('switches.prebidAppnexusInvcode', false);
+    config.set('ophan', { pageViewId: 'pvid' });
+    config.set('page.contentType', 'Article');
+    config.set('page.section', 'Magic');
+    config.set('page.edition', 'UK');
+};
+
+describe('getAppNexusInvCode', () => {
+    beforeEach(() => {
+        resetConfig();
+    });
+
+    afterEach(() => {
+        jest.resetAllMocks();
+        resetConfig();
+    });
+
+    test('should return the magic strings for mobile breakpoints', () => {
+        getBreakpointKey.mockReturnValue('M');
+        const invCodes = [
+            [[300, 250]],
+            [[300, 600]],
+            [[970, 250]],
+            [[728, 90]],
+        ].map(getAppNexusInvCode);
+
+        expect(invCodes).toEqual([
+            'Mmagic300x250',
+            'Mmagic300x600',
+            'Mmagic970x250',
+            'Mmagic728x90',
+        ]);
+    });
+
+    test('should return the magic strings for other breakpoints', () => {
+        getBreakpointKey.mockReturnValueOnce('T');
+        getBreakpointKey.mockReturnValueOnce('D');
+        const invCodes = [
+            [[300, 250]],
+            [[300, 600]],
+            [[970, 250]],
+            [[728, 90]],
+        ].map(getAppNexusInvCode);
+        expect(invCodes).toEqual([
+            'Dmagic300x250',
+            'Dmagic300x600',
+            'Dmagic970x250',
+            'Dmagic728x90',
+        ]);
+    });
+});
+
+describe('getAppNexusDirectPlacementId', () => {
+    beforeEach(() => {
+        resetConfig();
+        window.OzoneLotameData = { some: 'lotamedata' };
+    });
+
+    afterEach(() => {
+        jest.resetAllMocks();
+        resetConfig();
+        window.OzoneLotameData = undefined;
```

Comment:
#### 👉 Set up and tear down within each test block

window.OzoneLotameData gets set back to undefined